### PR TITLE
Fix out-of-sync ExpansionPanel animation

### LINKED ISSUE #5848: Expansion panel top margin animation may be out of sync.
The top of the expansion panel has a margin that animates from 16dp and 24dp. By playing with it in slow mode, the margin animation appears to sometimes finish first which it shouldn't; they should be in perfect sync.

### PR DESCRIPTION
**Pull request by morriskurz**
Increases the height of the AnimatedContainer in the ExpansionPanel class to match the IconButton height. This has no immediate effect on the height of the ExpansionPanel header, but smoothes the expansion / shrinking animation. See following GIFs for a visual explanation.

<details>
<summary>Before</summary>

</details>

<details>
<summary>After</summary>

</details>

Fixes #5848.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

Thank you @TahaTesser for the working example, it made my life a lot easier!

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by gspencergoog — COMMENTED** [review]
So, according to the [Material Design 2 spec](https://material.io/archive/guidelines/components/expansion-panels.html#expansion-panels-specs), the entire header is supposed to be 48 pixels high (i.e. kMinInteractiveDimension), including the expansion button.

It looks like when this component was authored, a padding of 16 was used around the expansion button, when it should have been 12. So instead of the changes you've made, I think the only thing that needs to change here is to make the padding on [this line](https://github.com/flutter/flutter/blob/3da9eee2f036ad80751b46c9df81b42b484b75ed/packages/flutter/lib/src/material/expansion_panel.dart#L360) be 12 instead of 16, which will make them both be 48 pixels high by default.

Ideally, some of those "magic" numbers would be moved to the top of the file as constants, and we'd have a test that makes sure that the header _and_ the collapse icon are both 48 pixels high.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Thanks for the PR!

File: packages/flutter/test/material/expansion_panel_test.dart [hunk 1]
```diff
@@ -1298,8 +1298,57 @@ void main() {
 
     // Padding is added to expanded header
     box = tester.renderObject(find.ancestor(of: find.byKey(firstPanelKey), matching: find.byType(AnimatedContainer)).first);
-    expect(box.size.height, equals(80.0)); // _kPanelHeaderCollapsedHeight + 32.0 (double default padding)
-    expect(box.size.width, equals(736.0));
+    expect(box.size.height, equals(80.0)); // _kPanelHeaderCollapsedHeight + 24.0 (double default padding)
+    expect(box.size.width, equals(744.0));
+  });
+
+  // Regression test for #5848
```
Comment (justinmc) [inline packages/flutter/test/material/expansion_panel_test.dart:1305]: Nit: This is usually formatted with the full URL like this:

```
// Regression test for https://github.com/flutter/flutter/issues/5848
```

File: packages/flutter/test/material/expansion_panel_test.dart [hunk 2]
```diff
@@ -1298,8 +1298,57 @@ void main() {
...
+        home: SingleChildScrollView(
+          child: SimpleExpansionPanelListTestWidget(
+            firstPanelKey: firstPanelKey,
+            canTapOnHeader: true,
+          ),
+        ),
+      ),
+    );
+
+    // The panel is closed
```
Comment (justinmc) [inline packages/flutter/test/material/expansion_panel_test.dart:1320]: Nit: Period at the end here.

Comment (justinmc) [reply packages/flutter/test/material/expansion_panel_test.dart:1320]: And in all of the other comments below.

**Review by justinmc — APPROVED** [review]
LGTM with nits 👍   Thanks for fixing this and for the gifs in the description!

**Comment by Hixie** [comment]
@morriskurz Thanks for your contribution! It looks like this is almost ready to land but has a few very minor fixes needed first. Would you have the time to do those changes so we could land it?

File: packages/flutter/test/material/expansion_panel_test.dart (hunk 1 above)
Comment (morriskurz) [reply packages/flutter/test/material/expansion_panel_test.dart:1305]: Thanks, I should have seen that in the other test cases!

File: packages/flutter/test/material/expansion_panel_test.dart (hunk 2 above)
Comment (morriskurz) [reply packages/flutter/test/material/expansion_panel_test.dart]: I copied the comment style from the other test cases to keep it consistent, but you are right - settings periods at the end of comments is part of the style guide! Thank you for the review.

**Comment by morriskurz** [comment]
@Hixie @gspencergoog @justinmc Thank you for the reviews and the effort you put in - I implemented the suggestions and should be ready to go!

**Review by justinmc — APPROVED** [review]
LGTM 👍 .  Thanks for fixing my nits.  I will add the autosubmit label to this and it will land soon 🚀

**Pull request merged**